src/util/compareThought.ts

```diff
@@ -221,3 +223,23 @@ export const compareThoughtByUpdated: ComparatorFunction<Thought> = (a: Thought,
 /** Compare two thoughts by their lastUpdated timestamp in descending order (newest first). Fall back to compareReasonable if created at the same time. */
 export const compareThoughtByUpdatedDescending: ComparatorFunction<Thought> = (a: Thought, b: Thought) =>
   compare(b.lastUpdated, a.lastUpdated) || compareReasonable(a.value, b.value)
+
+/** Compare two thoughts by their note value in ascending order, falling back to their value if notes are absent or equal. */
+export const compareThoughtByNote =
+  (state: State): ComparatorFunction<Thought> =>
+  (a: Thought, b: Thought) => {
+    const noteA = noteValue(state, a.id)
+    const noteB = noteValue(state, b.id)
+
+    return compareReasonable(noteA || a.value, noteB || b.value)
```

Comment:
I didn't think of thoughts that are missing notes when I create the original issue.

For mixed lists, could you sort all notes to the top, then sort by rank?

Please add a test for this case as well.

Thanks!